core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -1838,9 +1842,14 @@ public static long toLong(Object o) {
     return o instanceof Long ? (Long) o
         : o instanceof Number ? toLong((Number) o)
         : o instanceof String ? toLong((String) o)
+        : o instanceof java.util.Date ? toLong((java.util.Date) o)
         : (Long) cannotConvert(o, long.class);
```

Comment:
date to int ?

Reply:
Because `java.sql.Date/Time/TimeStamp` are the sub-classes of `java.util.Date`, and `SqlFunctions` also defines `toLong(java.util.Date ...)` methods to handle them.